src/klein/_app.py

```diff
@@ -429,7 +430,27 @@ def add(self, rule: Rule) -> None:
             self._url_map = _map_before_submount
             self._subroute_segments -= segments
 
-    def handle_errors(self, f_or_exception, *additional_exceptions):
+    @overload
+    def handle_errors(
+        self,
+        f_or_exception: KleinErrorHandler,
+        *additional_exceptions: Type[Exception],
+    ) -> Callable[[KleinErrorHandler], Callable]:
+        ...
+
+    @overload
+    def handle_errors(
+        self,
+        f_or_exception: Type[Exception],
+        *additional_exceptions: Type[Exception],
+    ) -> Callable[[KleinErrorHandler], Callable]:
+        ...
+
+    def handle_errors(
+        self,
+        f_or_exception: Union[KleinErrorHandler, Type[Exception]],
+        *additional_exceptions: Type[Exception],
+    ) -> Callable[[KleinErrorHandler], Callable]:
```

Comment:
Note I have the return type for the returned callable as a generic `Callable`.

This is in part because the `@modified` decorator is not type hinted, and I'm not really sure what it's supposed to be doing.  Perhaps @glyph can shed some light there.

In any case, I'm not especially worried about that, since I think that clients are unlikely to be effected by these specifics: the this checks the type of the callable passed in, which is what the client has control over.

Reply:
I *think* maybe at the time `@modified` was hinted, there were some Mypy bugs that prevented correctly modeling its argument?  Either that or I didn't quite know what I was doing.

In any case, I think that *now* we could do something like…

```
def modified(
    modification: str,
    original: Callable[..., C],
    modifier: Optional[[Callable[..., C]], Callable[..., C]] = None,
) -> Callable[[Callable[..., C]], Callable[..., C]]:
```

I vaguely recall, but do not have time to fully spelunk, that thanks to `_call`, we can't get more specific about argument list because `@modified` functions *do* necessarily cause a signature skew between `original` and `modifier`'s result (and therefore the parameterization of `modified`'s result), and Mypy still doesn't have support for procedural, or even declarative, dependent signature modification.  But I Think we *should* be able to do better than just `Callable`…

Reply:
Part of the difficulty here is that the doctoring for `@modified` needs work.  The `original` argument isn't documented, and I *think* that where the `modifier` argument is documented, that "this function" refers to `original`, but I'm not sure, and I haven't really had time to stare at it and figure all that out.

So I think we'll leave this plot twist for a future episode.

Reply:
> So I think we'll leave this plot twist for a future episode.

Happy to defer it, yes.